siliconcompiler/core.py

```diff
@@ -3106,29 +3048,45 @@ def node(self, flow, step, tool, task, index=0):
 
         The method modifies the following schema parameters:
 
-        ['flowgraph', flow, step, index, 'tool', tool]
-        ['flowgraph', flow, step, index, 'task', task]
-        ['flowgraph', flow, step, index, 'weight', metric]
+        * ['flowgraph', flow, step, index, 'tool', tool]
+        * ['flowgraph', flow, step, index, 'module', tool]
+        * ['flowgraph', flow, step, index, 'task', task]
+        * ['flowgraph', flow, step, index, 'weight', metric]
 
         Args:
             flow (str): Flow name
             step (str): Step name
             task (str): Task name, built in or tool specific
-            tool (str): Tool to associate with task.
+            tool (module): Tool to associate with task.
             index (int): Step index
 
         Examples:
-            >>> chip.node('asicflow', 'place', 'openroad', index=0)
-            Creates a task with step='place' and index=0 and binds it to the 'openroad' tool.
+            >>> import siliconcomiler.tools.openroad.openroad as openroad
+            >>> chip.node('asicflow', 'apr_place', openroad, 'place', index=0)
+            Creates a 'place' task with step='apr_place' and index=0 and binds it to the 'openroad' tool.
         '''
+
         if step in (Schema.GLOBAL_KEY, 'default'):
             self.error(f'Illegal step name: {step} is reserved')
             return
 
         index = str(index)
 
+        if (not inspect.ismodule(tool)):
+            self.error(f"{tool} is not a module and cannot be used to setup a tool.", fatal=True)
+
+        module_name = tool.__name__
+
+        import siliconcompiler as sc
```

Comment:
I assume putting this at the top causes cyclic import issues, but might be good to tack on a comment since mid-body imports always seem weird.

Reply:
yes